src/Network/Transport/TCP.hs

```diff
@@ -629,12 +663,189 @@ apiNewEndPoint transport =
     resolveMulticastGroupError =
       TransportError ResolveMulticastGroupUnsupported "Multicast not supported"
 
+-- | Specification of input queueing for a 'LocalEndPoint'
+data QPolicy prio = QPolicy {
+    -- | Determine a buffer constructor given an EndPointAddress.
+    --   Parametricity ensures the buffer is empty.
+    qpolicyBuffer :: forall t . EndPointAddress -> Buffer t
+    -- | Determine a priority for each EndPointAddress. Values which compare
+    --   lower give higher priority.
+  , qpolicyPriority :: EndPointAddress -> prio
```

Comment:
Please, document how EndPointAddress determines a queue. Which messages are placed in that queue?

Please, document how the priority influences processing of messages. Does `p0 < p1` mean that all messages from queue with priority `p0` need to be retrieved with `receive` before messages from `p1` are retrieved?